test/e2e/upgrades/topology_manager.go

```diff
@@ -0,0 +1,311 @@
+	cmd := tk.KubectlCmd("proxy", "-p", "0")
+	stdout, stderr, err := framework.StartCmdAndStreamOutput(cmd)
+	framework.ExpectNoError(err)
+	defer stdout.Close()
+	defer stderr.Close()
+	defer framework.TryKill(cmd)
+	buf := make([]byte, 128)
+	var n int
+	n, err = stdout.Read(buf)
+	framework.ExpectNoError(err)
+	output := string(buf[:n])
+	proxyRegexp := regexp.MustCompile("Starting to serve on 127.0.0.1:([0-9]+)")
+	match := proxyRegexp.FindStringSubmatch(output)
+	framework.ExpectEqual(len(match), 2)
+	port, err := strconv.Atoi(match[1])
+	framework.ExpectNoError(err)
+
+	ginkgo.By("http requesting node kubelet /configz")
+	endpoint := fmt.Sprintf("http://127.0.0.1:%d/api/v1/nodes/%s/proxy/configz", port, nodeName)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	req, err := http.NewRequest("GET", endpoint, nil)
+	framework.ExpectNoError(err)
+	req.Header.Add("Accept", "application/json")
+
+	var resp *http.Response
+	gomega.Eventually(func() bool {
+		resp, err = client.Do(req)
+		if err != nil {
+			framework.Logf("Failed to get /configz, retrying. Error: %v", err)
+			return false
+		}
+		if resp.StatusCode != 200 {
+			framework.Logf("/configz response status not 200, retrying. Response was: %+v", resp)
+			return false
+		}
+
+		return true
+	}, timeout, pollInterval).Should(gomega.Equal(true))
+	return resp
+}
+
+// Decodes the http response from /configz and returns a kubeletconfig.KubeletConfiguration (internal type).
+func decodeConfigz(resp *http.Response) (*kubeletconfig.KubeletConfiguration, error) {
+	// This hack because /configz reports the following structure:
+	// {"kubeletconfig": {the JSON representation of kubeletconfigv1beta1.KubeletConfiguration}}
+	type configzWrapper struct {
+		ComponentConfig kubeletconfigv1beta1.KubeletConfiguration `json:"kubeletconfig"`
+	}
+
+	configz := configzWrapper{}
+	kubeCfg := kubeletconfig.KubeletConfiguration{}
+
+	contentsBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(contentsBytes, &configz)
+	if err != nil {
+		return nil, err
+	}
+
+	err = scheme.Scheme.Convert(&configz.ComponentConfig, &kubeCfg, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &kubeCfg, nil
+}
+
+// cpusetCmd := fmt.Sprintf("grep Cpus_allowed_list /proc/self/status | cut -f2 && sleep 1d")
+func makePod(nodeName, namespace, podName, cmd string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      podName,
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{
+				{
+					Name:  podName,
+					Image: imageutils.GetE2EImage(imageutils.BusyBox),
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceName(v1.ResourceCPU):    resource.MustParse("2000m"),
```

Comment:
Can I know why this pod requires 2 CPU cores?
That can be too much on some environments, I feel.

Reply:
Two cores is the minimum amount of resources we can use to actually check the resources are aligned. But here we want to check if the topology manager works correctly across upgrades, hence we maybe can simplify a bit more and require just one core. I'll do some research and put the PR on hold until then.

Reply:
The topology manager is actually exercised using just one core, but I can't find a way to test it worked as intended with less than two cores, so I need to keep it this way.

Reply:
I think its fine that environments which require alignment testing verification during upgrades can reserve 2 cores.